arch/xtensa/src/esp32s3/esp32s3_spiflash_mtd.c

```diff
@@ -0,0 +1,807 @@
+ ****************************************************************************/
+
+/* ESP32-S3 SPI Flash device private data  */
+
+struct esp32s3_mtd_dev_s
+{
+  struct mtd_dev_s mtd;
+
+  /* SPI Flash data */
+
+  const struct spiflash_legacy_data_s **data;
+};
+
+/****************************************************************************
+ * Private Functions Prototypes
+ ****************************************************************************/
+
+/* MTD driver methods */
+
+static int esp32s3_erase(struct mtd_dev_s *dev, off_t startblock,
+                         size_t nblocks);
+static ssize_t esp32s3_read(struct mtd_dev_s *dev, off_t offset,
+                            size_t nbytes, uint8_t *buffer);
+static ssize_t esp32s3_read_decrypt(struct mtd_dev_s *dev,
+                                    off_t offset,
+                                    size_t nbytes,
+                                    uint8_t *buffer);
+static ssize_t esp32s3_bread(struct mtd_dev_s *dev, off_t startblock,
+                             size_t nblocks, uint8_t *buffer);
+static ssize_t esp32s3_bread_decrypt(struct mtd_dev_s *dev,
+                                     off_t startblock,
+                                     size_t nblocks,
+                                     uint8_t *buffer);
+static ssize_t esp32s3_write(struct mtd_dev_s *dev, off_t offset,
+                             size_t nbytes, const uint8_t *buffer);
+static ssize_t esp32s3_bwrite(struct mtd_dev_s *dev, off_t startblock,
+                              size_t nblocks, const uint8_t *buffer);
+static ssize_t esp32s3_bwrite_encrypt(struct mtd_dev_s *dev,
+                                      off_t startblock,
+                                      size_t nblocks,
+                                      const uint8_t *buffer);
+static int esp32s3_ioctl(struct mtd_dev_s *dev, int cmd,
+                         unsigned long arg);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const struct esp32s3_mtd_dev_s g_esp32s3_spiflash =
+{
+  .mtd =
+          {
+            .erase  = esp32s3_erase,
+            .bread  = esp32s3_bread,
+            .bwrite = esp32s3_bwrite,
+            .read   = esp32s3_read,
+            .ioctl  = esp32s3_ioctl,
+#ifdef CONFIG_MTD_BYTE_WRITE
+            .write  = esp32s3_write,
+#endif
+            .name   = "esp32s3_spiflash"
+          },
+  .data = &rom_spiflash_legacy_data,
+};
+
+static const struct esp32s3_mtd_dev_s g_esp32s3_spiflash_encrypt =
+{
+  .mtd =
+          {
+            .erase  = esp32s3_erase,
+            .bread  = esp32s3_bread_decrypt,
+            .bwrite = esp32s3_bwrite_encrypt,
+            .read   = esp32s3_read_decrypt,
+            .ioctl  = esp32s3_ioctl,
+#ifdef CONFIG_MTD_BYTE_WRITE
+            .write  = NULL,
+#endif
+            .name   = "esp32s3_spiflash_encrypt"
+          }
```

Comment:
```suggestion
          },
```
This missing comma is breaking the build.

Reply:
Thanks! Fixed!